### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by QuarpT**
## What does this change?

fix and remove flaky tests. don't hide renderer error message

## What is the value of this and can you measure success?

Tests don't randomly fail so much.

### CODE REVIEW

**Review by MatthewJWalls — APPROVED** [review]

File: common/app/rendering/core/Renderer.scala [hunk 1]
```diff
@@ -24,16 +22,17 @@ class Renderer(implicit actorSystem: ActorSystem, executionContext: ExecutionCon
...
-      .recover { case t: Throwable =>
-        val errorMessage = t.getLocalizedMessage.replaceAll("\u001B\\[[0-9]*m", "") // stripping terminal colors
-        log.error(errorMessage)
-        throw new RenderingException(errorMessage)
-      }
 
+    htmlF.failed.foreach { t =>
+      val errorMessage = Option(t.getLocalizedMessage).map(_.replaceAll("\u001B\\[[0-9]*m", "")).getOrElse("RendererError") // stripping terminal colors
+      log.error(errorMessage, t)
+    }
```
Comment (TBonnin) [inline common/app/rendering/core/Renderer.scala]: This means that the Play error message would show the error with the formatting included, which make it not very readable. Only an issue in Dev mode though.

File: common/test/package.scala [hunk 2]
```diff
@@ -73,6 +73,9 @@ trait ConfiguredTestSuite extends TestSuite with ConfiguredServer with Configure
 
 trait SingleServerSuite extends TestSuite with GuiceOneServerPerSuite with OneBrowserPerSuite with HtmlUnitFactory {
 
+  // Fixes `Failed to listen for HTTP on /0.0.0.0:19001`
+  System.setProperty("testserver.port", "0")
```
Comment (TBonnin) [inline common/test/package.scala]: what does this fix?

Comment (QuarpT) [reply common/test/package.scala]: it should fix these errors

```
sbt.ForkMain$ForkError: java.lang.RuntimeException: play.core.server.ServerListenException: Failed to listen for HTTP on /0.0.0.0:19001!
```

Comment (TBonnin) [reply common/test/package.scala]: Get that but why is anything trying to connect to `:19001`?

Comment (QuarpT) [reply common/test/package.scala]: It's the default listening port used by `GuiceOneServerPerSuite` when running the apps.

Comment (QuarpT) [reply common/test/package.scala]: see https://github.com/playframework/scalatestplus-play/blob/master/module/src/main/scala/org/scalatestplus/play/BaseOneServerPerSuite.scala

File: common/app/rendering/core/Renderer.scala (hunk 1 above)
Comment (QuarpT) [reply common/app/rendering/core/Renderer.scala]: It should be possible to turn off terminal formatting in bash if we need to in Dev

**Review by nicl — COMMENTED** [review]
Belated 👍 thanks for doing this.

**Comment by SiAdcock** [comment]
Omg thank you so much! 💕👌✅

**Pull request merged**